publisher.go

```diff
@@ -189,9 +181,9 @@ func (p *Publisher) Publish(ctx context.Context, msg interface{}, opts ...Publis
 	}
 
 	// finally, send
-	err = p.topicSender.Send(ctx, sbMsg)
+	err = p.topic.Send(ctx, sbMsg)
```

Comment:
topic.Send is a wrapper on Sender. it ensures the connection is healthy before sending the message.